src/streamlink/plugins/nrk.py

```diff
@@ -35,30 +15,73 @@
     )
 })
 
+_playable_schema = validate.Schema({
+    "playable": validate.all(
+        {
+            "assets": validate.all(
+                [{
+                    "url": validate.url(
+                        scheme="http",
+                        path=validate.any(
+                            validate.endswith(".m3u8"),
+                            validate.endswith(".mp3")
+                        ),
+                    ),
+                    "format": validate.all(validate.text),
+                }]
+            ),
+        }
+    ),
```

Comment:
Calling `validate.all` with just one argument is unnecessary.
Is it really necessary to limit the URL's `scheme` to `http`?
`validate.text` can nowadays be replaced by `str`, as it's an old py2 compat thing.

Reply:
I cleaned this up a bit, also wondering if it's ok to just remove the path validation. If the manifest returns an HLS playlist the `format` field explicitly states this, otherwise it should be any playable file stream. MIME type of the URL is also present in the manifest if needed, but not included here as it's not used.